src/ResourceManager/Profile/Commands.Profile/Models/RMProfileClient.cs

```diff
@@ -117,86 +118,118 @@ public bool TryRemoveContext(IAzureContext context)
+                    {
+                        Id = subscriptionId,
+                    };
                 }
+
+                else
+                {
+                    newSubscription = new AzureSubscription
+                    {
+                        Name = subscriptionName
+                    };
+                }
+
+                newTenant = new AzureTenant
+                {
+                    Id = tenantId
+                };
             }
-            // (tenant is not provided and subscription is present) OR
-            // (tenant is not provided and subscription is not provided)
+
             else
             {
-                var tenants = ListAccountTenants(account, environment, password, promptBehavior, promptAction)
-                    .Select(s => s.Id.ToString()).ToArray();
-                account.SetProperty(AzureAccount.Property.Tenants, null);
-                string accountId = null;
-
-                for (int i = 0; i < tenants.Count(); i++)
+                // (tenant and subscription are present) OR
+                // (tenant is present and subscription is not provided)
+                if (!string.IsNullOrEmpty(tenantId))
+                {
+                    var token = AcquireAccessToken(
+                        account,
+                        environment,
+                        tenantId,
+                        password,
+                        promptBehavior,
+                        promptAction);
+                    if (TryGetTenantSubscription(
+                        token,
+                        account,
+                        environment,
+                        tenantId,
+                        subscriptionId,
+                        subscriptionName,
+                        out newSubscription,
+                        out newTenant))
+                    {
+                        account.SetOrAppendProperty(AzureAccount.Property.Tenants, new[] { newTenant.Id.ToString() });
+                    }
+                }
+                // (tenant is not provided and subscription is present) OR
+                // (tenant is not provided and subscription is not provided)
+                else
                 {
-                    var tenant = tenants[i];
+                    var tenants = ListAccountTenants(account, environment, password, promptBehavior, promptAction)
+                        .Select(s => s.Id.ToString()).ToArray();
+                    account.SetProperty(AzureAccount.Property.Tenants, null);
+                    string accountId = null;
 
-                    IAzureTenant tempTenant;
-                    IAzureSubscription tempSubscription;
+                    for (int i = 0; i < tenants.Count(); i++)
+                    {
+                        var tenant = tenants[i];
 
-                    IAccessToken token = null;
+                        IAzureTenant tempTenant;
+                        IAzureSubscription tempSubscription;
```

Comment:
This should not be required anymore in C# 7.

You can now call a method and declare the `out` parameter in one statement.  E.g.,

```csharp
Function(input1, input1, out var output1, out var output2);
```

Reply:
Because C# 7 doesn't easily work in VS2015 (which we encourage people to use) I'm inclined not to put c# 7 code into the Profile module as this might cause a lot of people errors which we will then have to tell them how to fix.

Reply:
I wish there were a way to "dismiss" a comment without deleting it.